lib/utilities/retrieveData.js

```diff
@@ -47,24 +46,27 @@ export default function retrieveData(locations, retrieveDataFn, response, log) {
                 if (!responseDestroyed) {
                     _destroyResponse();
                 }
-                readable.emit('close');
+                readable.destroy();
                 return _next(responseErr);
             });
             // readable stream successfully consumed
-            readable.on('end', () => {
+            readable.once('end', () => {
+                readable.unpipe(response);
                 log.debug('readable stream end reached');
                 return _next();
             });
             // errors on server side with readable stream
-            readable.on('error', err => {
+            readable.once('error', err => {
                 log.error('error piping data from source');
+                _destroyResponse();
```

Comment:
why destroyResponse here and on line 69?  Won't it hit both?

Reply:
Same here : it was added before I got involved.
I agree with you, the logic of "_destroyResponse()" could be improved (ie test the boolean inside the function so that even if it's called twice it does not hurt ; even though I'm pretty sure calling it twice is fine anyway).

Reply:
This can be removed.

Reply:
ok, I'll remove it then